scripts/deployWFAIR.js

```diff
@@ -0,0 +1,76 @@
+/* ./scripts/deployWFAIR.js */
+import { transfers } from './utils/transfers';
+require('log-timestamp');
+const hre = require('hardhat');
+const toBN = hre.web3.utils.toBN;
+const fs = require('fs');
+
+const Q18 = (toBN('10')).pow(toBN('18'));
+
+// Load the deployment configuration file and set up constants
+const network = hre.hardhatArguments.network;
+console.log('Script is running on ' + network);
+const actionsFilepath = './scripts/' + network + '/logs/actions.json';
+const configFilepath = './scripts/' + network + '/deployWFAIR.config.json';
+
+let deployConfig;
+try {
+  deployConfig = JSON.parse(fs.readFileSync(configFilepath, 'utf-8'));
+  console.log('Configuration is retrieved from ' + configFilepath);
+} catch (err) {
+  console.error(err.message);
+}
+
+const actions = {};
+actions.transfers = [];
+console.log('Actions will be logged to ' + actionsFilepath);
+
+// In the config file we use WFAIR, but in the transactions we use wei <-- perhaps wei everywhere is better
+const WALLFAIR_TOTAL_SUPPLY = Q18.mul(toBN(deployConfig.wallfairTotalSupply)).toString();
+console.log('Total supply: ' + WALLFAIR_TOTAL_SUPPLY.replace(/\B(?=(\d{3})+(?!\d))/g, ',') + ' wei');
+
```

Comment:
check the following here
1. if enough ETH is present
2. if logs folder is available

[2021-09-20T11:21:53.462Z] Error: ENOENT: no such file or directory, open './scripts/localhost/logs/actions.json'
    at Object.openSync (fs.js:498:3)
    at Object.writeFileSync (fs.js:1524:35)
    at /home/rudolfix/src/EVNT-Token/scripts/deployWFAIR.js:68:8
    at processTicksAndRejections (internal/process/task_queues.js:95:5) {
  errno: -2,
  syscall: 'open',
  code: 'ENOENT',
  path: './scripts/localhost/logs/actions.json'
}

Reply:
1. I can't check at that point - I've done it in async function main after the accounts have been retrieved, using the same utils function as in deployTokenLocks.js.

2. Ditto - check if logs folder exists and create if needed is added but at end of file